src/system_info/system_info_network.cc

```diff
@@ -52,6 +52,20 @@ SysInfoNetwork::ToNetworkTypeString(SystemInfoNetworkType type) {
   return ret;
 }
 
+void SysInfoNetwork::SendUpdate() {
+  picojson::value output = picojson::value(picojson::object());
+  picojson::value data = picojson::value(picojson::object());
+
+  SetData(data);
+  system_info::SetPicoJsonObjectValue(output, "cmd",
+      picojson::value("SystemInfoPropertyValueChanged"));
+  system_info::SetPicoJsonObjectValue(output, "prop",
+      picojson::value("NETWORK"));
+  system_info::SetPicoJsonObjectValue(output, "data", data);
+
+  PostMessageToListeners(output);
+}
+
```

Comment:
@qjia7 abstract ````SendUpdate()```` , I didn't abstract ````OnTypeChanged````, because the argments are different, also with function logic.